content/05-expression.md

```diff
@@ -798,14 +818,184 @@ If during runtime the evaluation of `try-expression` causes a [`throw`](expressi
+
+Catch blocks are checked from top to bottom with the first one whose type is compatible with the thrown value being picked.
 
 This process has many similarities to the compile-time [unification](type-system-unification) behavior. However, since the check has to be done at runtime there are several restrictions:
 
 * The type must exist at runtime: [Class instances](types-class-instance), [enum instances](types-enum-instance), [abstract core types](types-abstract-core-type) and [Dynamic](types-dynamic).
 * Type parameters can only be [Dynamic](types-dynamic).
 
+#### Wildcard catch
+
+##### Haxe 3.* and Haxe 4.0
+
+Prior to Haxe 4.1.0 the only way to catch all exceptions is by using `Dynamic` as the catch type.
+To get a string representation of the exception `Std.string(e)` could be used. 
+```haxe
+try {
+  doSomething();
+} catch(e:Dynamic) {
+  // All exceptions will be caught here
+  trace(Std.string(e));
+}
+```
+
+#### Since Haxe 4.1
+
+Instead of `Dynamic` it's possible (and recommended) to use `haxe.Exception` type for wildcard catches:
+```haxe
+try {
+  doSomething();
+} catch(e:haxe.Exception) {
+  //All exceptions will be caught here
+  trace(e.message);
+}
+```
+
+#### Exception stack 
+
+##### Haxe 3.* and Haxe 4.0
+
+The exception call stack is available via `haxe.CallStack.exceptionStack()` inside of a `catch` block:
+```haxe
+try {
+  doSomething();
+} catch(e:Dynamic) {
+  var stack = haxe.CallStack.exceptionStack();
+  trace(haxe.CallStack.toString(stack));
+}
+```
+
+##### Since Haxe 4.1
+
+If the catch type is `haxe.Exception` or one of its descendants, then the exception stack is available in the `stack` property of the exception instance.
+```haxe
+try {
+  doSomething();
+} catch(e:haxe.Exception) {
+  trace(e.stack);
+}
+```
+
+#### Rethrowing exceptions
+
+##### Since Haxe 4.1
+
+Even if an instance of `haxe.Exception` is thrown again, it still preserves all the original information, including the stack.
+```haxe
+import haxe.Exception;
+
+class Main {
+  static function main() {
+    try {
+      try {
+        doSomething();
+      } catch(e:Exception) {
+        trace(e.stack);
+        throw e; //rethrow
+      }
+    } catch(e:Exception) {
+      trace(e.stack);
+    }
+  }
+
+  static function doSomething() {
+    throw new Exception('Terrible error');
+  }
+}
+```
+This example being executed with `haxe --main Main --interp` would print something like this:
+```
+Main.hx:13: 
+Called from Main.doSomething (Main.hx line 11 column 15)
+Called from Main.main (Main.hx line 5 column 5)
+Main.hx:17: 
+Called from Main.doSomething (Main.hx line 11 column 15)
+Called from Main.main (Main.hx line 5 column 5)
+```
+
+If `catch(e:haxe.Exception)` catches a native exception, then rethrowing `e` throws that native exception instead of a `haxe.Exception` instance.
```

Comment:
Hmm, now I'm a bit confused because this would imply that you _can't_ catch any exception with `catch (e:haxe.Exception)`.

Reply:
I don't see why? Maybe my English skills fail me )

Reply:
It says that a native exception is thrown _instead_ of a `haxe.Exception` instance. This tells me that the native exception is not a `haxe.Exception` instance, and it follows that it can't be caught as such.

Reply:
Would adding "but it still could be caught with `catch(e:haxe.Exception)`" help?